iml-agent/src/action_plugins/lamigo.rs

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2019 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::{agent_error::ImlAgentError, daemon_plugins, env};
+use futures::future::TryFutureExt;
+use std::collections::HashMap;
+use std::path::Path;
+use tokio::fs;
+use tokio::io::AsyncWriteExt;
+
+#[derive(serde::Deserialize, structopt::StructOpt, Clone, Debug)]
+pub struct Config {
+    #[structopt(long)]
+    /// File system name
+    fs: String,
+
+    #[structopt(long)]
+    /// MDT device that provides changelogs
+    mdt: String,
+
+    #[structopt(long)]
+    /// The user on which behalf lamigo consumes changelogs from the MDT device
+    user: String,
+
+    #[structopt(long)]
+    /// Fast OST pool name
+    hot_pool: String,
+
+    #[structopt(long)]
+    /// Slow OST pool name
+    cold_pool: String,
+
+    #[structopt(long)]
+    /// Interval for lamigo to wait before replicating a closed file (in seconds)
+    min_age: u32,
+
+    #[structopt(long)]
+    /// Lustre device to be mounted, e.g. `192.168.0.100@tcp0:/spfs`
+    lustre_device: String,
+
+    #[structopt(long)]
+    /// IML mailbox name, e.g. `mailbox1`, where `lamigo` will write FIDs
+    mailbox: String,
+}
+
+fn format_lamigo_unit_contents(c: &Config, iml_socket: &str) -> String {
+    format!(
+        "[Unit]\n\
+         Description=Run lamigo service\n\
+         [Service]\n\
+         ExecStart=/usr/bin/lamigo \
+         -m {fs}-{mdt} \
+         -u {user} \
+         -s {hot_pool} \
+         -t {cold_pool} \
+         -a {min_age} \
+         --iml-socket {iml_socket}\n\
+         ",
+        fs = c.fs,
+        mdt = c.mdt,
+        cold_pool = c.cold_pool,
+        hot_pool = c.hot_pool,
+        min_age = c.min_age,
+        user = c.user,
+        iml_socket = iml_socket
+    )
+}
+
+pub async fn create_lamigo_service_unit(c: Config) -> Result<(), ImlAgentError> {
+    let path_fmt = env::get_var("LAMIGO_UNIT_PATH");
+    let iml_socket = daemon_plugins::postoffice::socket_name(&c.mailbox); // wants SOCK_DIR
```

Comment:
Not related to this patch, but feels like this should be in `env.rs` It seems strange to reach into another plugin to get some info.